adaptors/dummy/test/e2e/usability/usability_test.go

```diff
@@ -0,0 +1,552 @@
+package usability_test
+
+import (
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	apiv1alpha1 "github.com/rancher/octopus/adaptors/dummy/api/v1alpha1"
+	edgev1alpha1 "github.com/rancher/octopus/api/v1alpha1"
+	"github.com/rancher/octopus/pkg/util/object"
+	"github.com/rancher/octopus/test/util/content"
+	"github.com/rancher/octopus/test/util/exec"
+	"github.com/rancher/octopus/test/util/node"
+)
+
+var (
+	deviceLink edgev1alpha1.DeviceLink
+)
+
+var _ = Describe("verify usability", func() {
+
+	BeforeEach(func() {
+		// create device link
+		deployDummyDeviceLink()
+	})
+
+	AfterEach(func() {
+		// delete device link, ignore error
+		_ = k8sCli.DeleteAllOf(testCtx, &edgev1alpha1.DeviceLink{}, client.InNamespace(deviceLink.Namespace))
+	})
+
+	Context("edit dummy device link spec", func() {
+		Specify("if invalid node spec", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
```

Comment:
Looks like we gave a device link which is Connected, please clarify the description, and adjust the other same places.

Reply:
updated